packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -145,23 +148,21 @@ function getAllAttributesInRelation({ relations }: Relations): Attributes[] {
 
 // Function checks if attribute is valid and has location data in it.
 // Since have observed noisy data before so this function is required.
-function isValidAttribute(attr: Attributes): boolean {
-  if (attr && attr.location && attr.location.begin && attr.location.end) {
-    return true;
-  }
-  return false;
+function hasLocationData(attr: Attributes): boolean {
+  return !!(
+    attr &&
+    attr.location &&
+    typeof attr.location.begin !== 'undefined' &&
+    attr.location.end
+  );
 }
 
 export default function transformEnrichment(enrichedHtml: EnrichedHtml[]): EnrichedHtml[] {
-  const newEnrichedHtml = { ...enrichedHtml };
-
-  if (newEnrichedHtml && newEnrichedHtml[0]) {
-    const enrichmentName = getEnrichmentName(newEnrichedHtml[0]);
-    const enrichment = newEnrichedHtml[0][enrichmentName];
-    if (enrichment) {
-      modelMapping[enrichmentName](enrichment, enrichmentName);
-    }
+  if (enrichedHtml && enrichedHtml[0]) {
+    const enrichmentName = getEnrichmentName(enrichedHtml[0]);
+    const enrichment = enrichedHtml[0][enrichmentName];
+    const updatedEnrichment = getUpdatedEnrichment(enrichment, enrichmentName);
+    enrichedHtml[0][enrichmentName] = updatedEnrichment;
```

Comment:
As I noted above, we should probably make the input arguments immutable.

Reply:
https://github.com/watson-developer-cloud/discovery-components/pull/69/files#r377243755